shell/browser/browser.cc

```diff
@@ -122,6 +122,12 @@ std::string Browser::GetName() const {
 
 void Browser::SetName(const std::string& name) {
   OverrideApplicationName(name);
+  if (!is_ready()) {
+    // DIR_USER_DATA, DIR_USER_CACHE and DIR_APP_LOGS depend
+    // on application name.
+    // We have to force re-computation
+    base::PathService::InvalidateCache();
+  }
```

Comment:
The intent and implementation is fine but that `if (!is_ready())` line is going to be confusing to future maintainers. IMO the comment should take it into account. What would you think of

```cc
// Some of the paths in `app.getPath()` are based on `name` and should
// update when `name` is changed before `app.ready` is fired. So if app
// isn't ready yet, invalidte the PathService cache to force a refresh.
if (!is_ready()) {
  base::PathService::InvalidateCache();
}
```